src/query/interpreter.cpp

```diff
@@ -2197,11 +2204,17 @@ PreparedQuery PrepareIndexQuery(ParsedQuery parsed_query, bool in_explicit_trans
       index_notification.title = fmt::format("Dropped index on label {} on properties {}.", index_query->label_.name,
                                              utils::Join(properties_string, ", "));
       // TODO: not just storage + invalidate_plan_cache. Need a DB transaction (for replication)
-      handler = [dba, label, properties_stringified = std::move(properties_stringified),
-                 label_name = index_query->label_.name, properties = std::move(properties),
+      handler = [dba, index_type = index_query->type_, label, index_name,
+                 properties_stringified = std::move(properties_stringified), label_name = index_query->label_.name,
+                 properties = std::move(properties),
                  invalidate_plan_cache = std::move(invalidate_plan_cache)](Notification &index_notification) {
         MG_ASSERT(properties.size() <= 1U);
-        auto maybe_index_error = properties.empty() ? dba->DropIndex(label) : dba->DropIndex(label, properties[0]);
+        utils::BasicResult<storage::StorageIndexDefinitionError, void> maybe_index_error{};
```

Comment:
You can use std::optional here as wrapper, and then emplace inside `maybe_index_error`, as it avoids creation of object until line 2214 or 2216

Reply:
Added